### ISSUE TITLE
Squeeze error in InceptionV1

### ISSUE THREAD

**Issue by hegman12** [tensorflow/lucid#101]
Hi There,
I am getting an error when i tried to use the objective `class_logit ` in lucid. Blow is a link to the colab notebook to reproduce the error. Could you please point me where am i going wrong?

https://colab.research.google.com/drive/1yE6fCkEvNMRV-nsmJCT2EzfRv7CCXwaA

**Comment by ludwigschubert** [comment]
Multiple things going on here:

The tensor you're passing to `class_logits` comes after a spatial squeeze op, which expects precise spatial dimensions. By default (and we'll reconsider that) we add transformation robustness, which requires padding, which screws up that shape. 

Other than explicitly setting `transforms=[]` to prevent from lucid mucking up your image shape, you can (and probably should!) also optimize for activations before the spatial squeeze op. (Pre-softmax optimization often leads to better results anyway, as explained in [Feature Visualization](https://distill.pub/2017/feature-visualization/).) In your example, a candidate tensor could be `InceptionV1/Logits/Conv2d_0c_1x1/BiasAdd`.

Finally, feature vis of classes/logits doesn't by default produce "good" (as in "human-interpretable") results atm, so please don't expect any magic—there are probably multiple steps we'd need to take before getting good results from these, and it hasn't been a strong focus for lucid so far. If you end up exploring how to better optimize these, please [let us know](https://join.slack.com/t/distillpub/shared_invite/enQtMzg1NzU3MzEzMTg3LWJkNmQ4Y2JlNjJkNDlhYTU2ZmQxMGFkM2NiMTI2NGVjNzJkOTdjNTFiOGZmNDBjNTEzZGUwM2U0Mzg4NDAyN2E) about your progress!

**Comment by ludwigschubert** [comment]
Closing for now, but if my recommendations don't help, feel free to reopen! :-)

**Comment by hegman12** [comment]
Thanks :) I will try your recommendations!
